### PR TITLE
infra: run tests.yml workflow on ubuntu runners also for rhel branches 

### PR DESCRIPTION
**Pull request by adamkankovsky**

### CODE REVIEW

**Review by KKoukiou — COMMENTED** [review]
Please rephrase commit message to:

infra: run tests.yml workflow on ubuntu runners also for rhel branches 

And the commit description can be also added. Something like:

rhel branches use public centos container images which can be pulled and used also on public runners.

**Review by KKoukiou — COMMENTED** [review]
Please make the correspondant rhel-9 and rhel-10 branch changes to verify that this actually works before merging :)

**Review by KKoukiou — COMMENTED** [review]
This change is good, but the PR title needs adjustment, as it refers only to the first commit.

File: .github/workflows/push-tests.yml.j2
Comment (KKoukiou) [inline .github/workflows/push-tests.yml.j2:7]: We actually dont hardcode branch names. You need to generate it. 

See {% for branch in supported_branches ...

**Review by KKoukiou — CHANGES_REQUESTED** [review]

File: .github/workflows/push-tests.yml
Comment (KKoukiou) [inline .github/workflows/push-tests.yml:11]: Also I dont understand the need for this workflow.

We dont allow pushes to master branch from the settings, and we generally dont push directly - we always create a PR. I think this workflow can go away - since there is another workflow covering the tests execution on PRs.

**Review by KKoukiou — CHANGES_REQUESTED** [review]

File: .github/workflows/push-tests.yml
Comment (KKoukiou) [reply .github/workflows/push-tests.yml:11]: OK I see from the git history that this was introduced for codecov reasons. 
Do we run codecov on all branches @jkonecny12 ?

File: .github/workflows/push-tests.yml.j2
Comment (jkonecny12) [reply .github/workflows/push-tests.yml.j2:7]: Or wait on https://github.com/rhinstaller/anaconda/pull/5598 which will add matrix builder. Both options are fine.

File: .github/workflows/push-tests.yml
Comment (jkonecny12) [reply .github/workflows/push-tests.yml:11]: Yes, we do https://app.codecov.io/gh/rhinstaller/anaconda/tree/rhel-10.

File: .github/workflows/push-tests.yml.j2
Comment (jkonecny12) [inline .github/workflows/push-tests.yml.j2:1]: Why don't we completely remove this template file. It's not used now.

**Review by jkonecny12 — CHANGES_REQUESTED** [review]
Please remove unused template files.

**Comment by jkonecny12** [comment]
Looks good to me but I wonder if we want to remove template file together with the unused workflow. Maybe that is the correct path. At least I don't see an issue with this approach.

File: .github/workflows/tests.yml
Comment (jkonecny12) [inline .github/workflows/tests.yml:21]: We are just discussing this on other PR. But I don't think it should block this PR.

**Review by jkonecny12 — APPROVED** [review]
Looks good, however, these changes should be merged to `master` first.

File: .github/workflows/push-tests.yml.j2
Comment (KKoukiou) [reply .github/workflows/push-tests.yml.j2:1]: @adamkankovsky he did not mean to remote the workflow, he meant to remove the template file. The actual workflow can stay. Please do this in another PR, in order to keep this moving.

File: .github/workflows/tests.yml
Comment (KKoukiou) [reply .github/workflows/tests.yml:21]: OK let's merge this. But please take care of keeping such changes split up, not hidden inside random commits @adamkankovsky

**Review by KKoukiou — APPROVED** [review]

**Comment by KKoukiou** [comment]
Testing farm is taking ages - merging - the installability test from testing farm cant be affected by this.

**Pull request merged**
